theano/tensor/blas.py

```diff
@@ -250,10 +270,11 @@ def perform(self, node, inputs, out_storage):
             out = numpy.dot(A, x)
             if alpha != 1:
                 out *= alpha
-            if beta != 1:
-                out += beta * y
-            else:
-                out += y
+            if beta != 0:
+                if beta != 1:
+                    out += beta * y
+                else:
+                    out += y
```

Comment:
This have the inconvenience of hiding nan. We want to prevent false nan, but we don't want to hide real nan.

I think it is fine to hide them if blas isn't good. But here, we don't do that. I would just remove this change.